onedal/spmd/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,123 @@
+from ...datatypes import _convert_to_supported, from_table, to_table
+from ...decomposition import IncrementalPCA as IncrementalPCA_nonSPMD
+from ...utils import _check_array
+from .._base import BaseEstimatorSPMD
+
+
+class IncrementalPCA(BaseEstimatorSPMD, IncrementalPCA_nonSPMD):
+    def _reset(self):
+        self._partial_result = super(IncrementalPCA_nonSPMD, self)._get_backend(
+            "decomposition", "dim_reduction", "partial_train_result"
+        )
+        if hasattr(self, "components_"):
+            del self.components_
+
+    @support_usm_ndarray()
+    def partial_fit(self, X, queue):
+        """Incremental fit with X. All of X is processed as a single batch.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        y : Ignored
+            Not used, present for API consistency by convention.
+
+        check_input : bool, default=True
+            Run check_array on X.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        X = _check_array(X)
+        n_samples, n_features = X.shape
+
+        first_pass = not hasattr(self, "components_")
+        if first_pass:
+            self.components_ = None
+            self.n_samples_seen_ = n_samples
+            self.n_features_in_ = n_features
+        else:
+            self.n_samples_seen_ += n_samples
+
+        if self.n_components is None:
+            if self.components_ is None:
+                self.n_components_ = min(n_samples, n_features)
+            else:
+                self.n_components_ = self.components_.shape[0]
+        else:
+            self.n_components_ = self.n_components
+
+        if not hasattr(self, "_queue"):
+            self._queue = queue
+
+        policy = super(IncrementalPCA_nonSPMD, self)._get_policy(queue, X)
+        X = _convert_to_supported(policy, X)
+
+        if not hasattr(self, "_dtype"):
+            self._dtype = get_dtype(X)
+            self._params = self._get_onedal_params(X)
+
+        X_table = to_table(X)
+        self._partial_result = super(IncrementalPCA_nonSPMD, self)._get_backend(
+            "decomposition",
+            "dim_reduction",
+            "partial_train",
+            policy,
+            self._params,
+            self._partial_result,
+            X_table,
+        )
+        return self
+
+    def _create_model(self):
+        m = super(IncrementalPCA_nonSPMD, self)._get_backend(
+            "decomposition", "dim_reduction", "model"
+        )
+        m.eigenvectors = to_table(self.components_)
+        m.means = to_table(self.mean_)
+        if self.whiten:
+            m.eigenvalues = to_table(self.explained_variance_)
+        self._onedal_model = m
+        return m
+
+    def predict(self, X, queue=None):
```

Comment:
would predict need to be in this file? because we don't support PCA spmd predict and predict would be same as non-incremental?

Reply:
this is exactly the reason predict is necessary in this file. since we don't support spmd predict, it should be explicitly specified here that we call non-SPMD backend in predict

Reply:
Wouldn't it default to existing https://github.com/intel/scikit-learn-intelex/blob/main/onedal/decomposition/pca.py#L130-L139 ?

Reply:
it would call spmd get_policy and get_backend and it leads to fail